CircleCI: D402 First line should not be the function's "signature"

Recent issue on the master branch, most likely due to a flake8 (plugin) version change:

```
$ pre-commit run -a

Check that executables have shebangs.....................................Passed
Check JSON...............................................................Passed
Debug Statements (Python)................................................Passed
Fix End of Files.........................................................Passed
Mixed line ending........................................................Passed
black....................................................................Passed
flake8...................................................................Failed
- hook id: flake8
- exit code: 1

Bio/SeqFeature.py:1676:1: D402 First line should not be the function's "signature"
Bio/SeqFeature.py:1812:1: D402 First line should not be the function's "signature"
Bio/SeqFeature.py:1930:1: D402 First line should not be the function's "signature"
Bio/SeqFeature.py:1995:1: D402 First line should not be the function's "signature"
Bio/SeqFeature.py:2071:1: D402 First line should not be the function's "signature"
Bio/Phylo/BaseTree.py:465:1: D402 First line should not be the function's "signature"
Bio/GenBank/__init__.py:726:1: D402 First line should not be the function's "signature"
Bio/GenBank/__init__.py:1496:1: D402 First line should not be the function's "signature"

blacken-docs.............................................................Passed
rstcheck.................................................................Passed
doc8.....................................................................Passed
Enforce https://doi.org/ style...........................................Passed
```

I think this started while CircleCI was not running promptly, and we merged without it passing.

--- a/.pre-commit-config.yaml
+++ b/.pre-commit-config.yaml
@@ -16,7 +16,7 @@ repos:
     -   id: black
         args: [--check]
 -   repo: https://gitlab.com/pycqa/flake8
-    rev: 3.7.9
+    rev: 3.9.1
     hooks:
     -   id: flake8
         additional_dependencies: [
@@ -26,8 +26,8 @@ repos:
             'flake8-docstrings',
             'flake8-implicit-str-concat',
             'flake8-quotes',
-            'flake8-rst-docstrings',
-            'pydocstyle>=5.0.0',
+            'flake8-rst-docstrings>=0.2.3',
+            'pydocstyle>=6.0.0',
         ]
 -   repo: https://github.com/asottile/blacken-docs
     rev: v1.6.0

--- a/Bio/GenBank/__init__.py
+++ b/Bio/GenBank/__init__.py
@@ -722,8 +722,11 @@ class _FeatureConsumer(_BaseGenBankConsumer):
         """
         self._seq_type = type.strip()
 
-    def topology(self, topology):  # noqa: D402
-        """Validate and record sequence topology (linear or circular as strings)."""
+    def topology(self, topology):
+        """Validate and record sequence topology.
+
+        The topology argument should be "linear" or "circular" (string).
+        """
         if topology:
             if topology not in ["linear", "circular"]:
                 raise ParserFailureError(
@@ -1490,10 +1493,11 @@ class _RecordConsumer(_BaseGenBankConsumer):
 
             self.data.molecule_type = mol_type
 
-    def topology(
-        self, topology
-    ):  # noqa: D402  # flake8 thinks this line is a function signature. It ain't
-        """Validate and record sequence topology (linear or circular as strings)."""
+    def topology(self, topology):
+        """Validate and record sequence topology.
+
+        The topology argument should be "linear" or "circular" (string).
+        """
         if topology:
             if topology not in ["linear", "circular"]:
                 raise ParserFailureError(

--- a/Bio/Phylo/BaseTree.py
+++ b/Bio/Phylo/BaseTree.py
@@ -474,7 +474,7 @@ class TreeMixin:
             instances in the tree, and values are the distance from the root to
             each clade (including terminals).
 
-        """
+        """  # noqa: D402
         if unit_branch_lengths:
             depth_of = lambda c: 1  # noqa: E731
         else:

--- a/Bio/SeqFeature.py
+++ b/Bio/SeqFeature.py
@@ -1673,7 +1673,7 @@ class UnknownPosition(AbstractPosition):
 
     @property
     def extension(self):  # noqa: D402
-        """Legacy attribute to get extension (zero) as integer (OBSOLETE)."""
+        """Legacy attribute to get extension (zero) as integer (OBSOLETE)."""  # noqa: D402
         return 0
 
     def _shift(self, offset):
@@ -1809,7 +1809,7 @@ class WithinPosition(int, AbstractPosition):
 
     @property
     def extension(self):  # noqa: D402
-        """Legacy attribute to get extension (from left to right) as an integer (OBSOLETE)."""
+        """Legacy attribute to get extension (from left to right) as an integer (OBSOLETE)."""  # noqa: D402
         return self._right - self._left
 
     def _shift(self, offset):
@@ -1927,7 +1927,7 @@ class BetweenPosition(int, AbstractPosition):
 
     @property
     def extension(self):  # noqa: D402
-        """Legacy attribute to get extension (from left to right) as an integer (OBSOLETE)."""
+        """Legacy attribute to get extension (from left to right) as an integer (OBSOLETE)."""  # noqa: D402
         return self._right - self._left
 
     def _shift(self, offset):
@@ -1992,7 +1992,7 @@ class BeforePosition(int, AbstractPosition):
 
     @property
     def extension(self):  # noqa: D402
-        """Legacy attribute to get extension (zero) as integer (OBSOLETE)."""
+        """Legacy attribute to get extension (zero) as integer (OBSOLETE)."""  # noqa: D402
         return 0
 
     def __repr__(self):
@@ -2068,7 +2068,7 @@ class AfterPosition(int, AbstractPosition):
 
     @property
     def extension(self):  # noqa: D402
-        """Legacy attribute to get extension (zero) as integer (OBSOLETE)."""
+        """Legacy attribute to get extension (zero) as integer (OBSOLETE)."""  # noqa: D402
         return 0
 
     def __repr__(self):

